lib/bots/keyframebot.ts

```diff
@@ -0,0 +1,683 @@
+                    },
+                    method: this.githubApi.gitdata.createTree
+                });
+            }
+        }).then((newTree: GithubApiTypes.Tree) => {
+            newTreeSha = newTree.sha;
+
+            // Get the last commit for the branch.
+            return this.dispatchToEmitter(this.githubEmitterName, {
+                data: {
+                    owner,
+                    repo,
+                    sha: branchSha
+                },
+                method: this.githubApi.repos.getCommit
+            });
+        }).then((lastCommit: GithubApiTypes.Commit) => {
+            // We have new tree object, we now want to create a new commit referencing it.
+            return this.dispatchToEmitter(this.githubEmitterName, {
+                data: {
+                    message: `Update keyframe from product version ${version} on behalf of Resin user ${user}.`,
+                    owner,
+                    parents: [ lastCommit.sha ],
+                    repo,
+                    tree: newTreeSha
+                },
+                method: this.githubApi.gitdata.createCommit
+            });
+        }).then((commit: GithubApiTypes.Commit) => {
+            // Update the branch to include the new commit SHA, so the head points to our new
+            // keyframe.
+            commitSha = commit.sha;
+            return this.dispatchToEmitter(this.githubEmitterName, {
+                data: {
+                    force: false,
+                    owner,
+                    ref: `heads/${branchName}`,
+                    repo,
+                    sha: commitSha
+                },
+                method: this.githubApi.gitdata.updateReference
+            });
+        }).then(() => {
+            return this.dispatchToEmitter(this.githubEmitterName, {
+                data: {
+                    message: version,
+                    object: commitSha,
+                    owner,
+                    repo,
+                    tag: version,
+                    tagger: {
+                        name: process.env.KEYFRAMEBOT_NAME
+                    },
+                    type: 'commit'
+                },
+                method: this.githubApi.gitdata.createTag
+            });
+        }).then((newTag: GithubApiTypes.Tag) => {
+            // Finally tag this with the version of the keyframe.
+            return this.dispatchToEmitter(this.githubEmitterName, {
+                data: {
+                    owner,
+                    ref: `refs/tags/${version}`,
+                    repo,
+                    sha: newTag.sha
+                },
+                method: this.githubApi.gitdata.createReference
+            });
+        }).return(branchName);
+    }
+
+    /**
+     * Reports an error to the console and as a Github comment..
+     *
+     * @param error The error to report.
+     */
+    private reportError(error: GithubError | HTTPError): void {
+        // Log to console.
```

Comment:
Comment above says "to the console and as a Github comment" but comment here just says "Log to console" ?